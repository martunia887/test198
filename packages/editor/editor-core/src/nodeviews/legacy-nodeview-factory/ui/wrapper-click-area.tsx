import * as React from 'react';
import { PureComponent, ComponentClass, StatelessComponent } from 'react';
import { EditorView } from 'prosemirror-view';
import styled from 'styled-components';
import {
  ReactNodeViewState,
  stateKey as reactNodeViewPlugin,
} from '../../../plugins/base/pm-plugins/react-nodeview';
import { setNodeSelection } from '../../../utils';
import { ProsemirrorGetPosHandler } from '../../types';

export interface ReactNodeViewComponents {
  [key: string]: ComponentClass<any> | StatelessComponent<any>;
}

export interface ClickWrapperProps {
  getPos: ProsemirrorGetPosHandler;
  pluginState: ReactNodeViewState;
  view: EditorView;

  onSelection?: (selected: boolean) => void;
}

const BlockWrapper = styled.div`
  width: 100%;
`;
BlockWrapper.displayName = 'BlockWrapperClickArea';

const InlineWrapper = styled.span``;
InlineWrapper.displayName = 'InlineWrapperClickArea';

interface State {
  selected: boolean;
}

export default function wrapComponentWithClickArea<
  T extends Partial<ClickWrapperProps> & { selected: boolean }
>(
  ReactComponent: React.ComponentType<T>,
  inline?: boolean,
): React.ComponentClass<T & ClickWrapperProps> {
  return class WrapperClickArea extends PureComponent<
    T & ClickWrapperProps,
    State
  > {
    state: State = { selected: false };

    componentDidMount() {
      const { pluginState } = this.props;
      pluginState.subscribe(this.handleDocumentSelectionChange);
    }

    componentWillUnmount() {
      const { pluginState } = this.props;
      pluginState.unsubscribe(this.handleDocumentSelectionChange);
    }

    render() {
      const Wrapper = inline ? InlineWrapper : BlockWrapper;
      return (
        <Wrapper onClick={this.onClick}>
          <ReactComponent {...this.props} selected={this.state.selected} />
        </Wrapper>
      );
    }

    private handleDocumentSelectionChange = (
      fromPos: number,
      toPos: number,
    ) => {
      const { getPos, onSelection } = this.props;
      const nodePos = getPos();

      const selected = nodePos >= fromPos && nodePos < toPos;

      const oldSelected = this.state.selected;
      this.setState({ selected }, () => {
        if (onSelection && selected !== oldSelected) {
          onSelection(selected);
        }
      });
    };

    private onClick = () => {
      const { getPos, view } = this.props;
      setNodeSelection(view, getPos());
    };
  };
}

export function applySelectionAsProps<T extends ClickWrapperProps>(
  Component: React.ComponentType<T>,
) {
  return class extends React.PureComponent<T, {}> {
    render() {
      return (
        <Component
          {...this.props}
          pluginState={reactNodeViewPlugin.getState(this.props.view.state)}
        />
      );
    }
  };
}
