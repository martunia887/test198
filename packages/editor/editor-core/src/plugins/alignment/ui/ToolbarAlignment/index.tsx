import * as React from 'react';
import { defineMessages } from 'react-intl';
import ExpandIcon from '@atlaskit/icon/glyph/chevron-down';
import EditorAlignCenterIcon from '@atlaskit/icon/glyph/editor/align-center';
import EditorAlignLeftIcon from '@atlaskit/icon/glyph/editor/align-left';
import EditorAlignRightIcon from '@atlaskit/icon/glyph/editor/align-right';

import Alignment from '../../../../ui/Alignment';
import Dropdown from '../../../../ui/Dropdown';
import ToolbarButton from '../../../../ui/ToolbarButton';
import { AlignmentPluginState, AlignmentState } from '../../pm-plugins/main';

import {
  TriggerWrapper,
  Separator,
  Wrapper,
  ExpandIconWrapper,
} from './styles';

export const messages = defineMessages({
  alignment: {
    id: 'fabric.editor.alignment',
    defaultMessage: 'Alignment',
    description: 'Aligns text',
  },
});

export interface State {
  isOpen: boolean;
}

export interface Props {
  pluginState: AlignmentPluginState;
  changeAlignment: (align: AlignmentState) => void;
  popupsMountPoint?: HTMLElement;
  popupsBoundariesElement?: HTMLElement;
  popupsScrollableElement?: HTMLElement;
  isReducedSpacing?: boolean;
  disabled?: boolean;
}

class AlignmentToolbar extends React.Component<Props, State> {
  state: State = {
    isOpen: false,
  };

  render() {
    const { isOpen } = this.state;
    const {
      popupsMountPoint,
      popupsBoundariesElement,
      popupsScrollableElement,
      isReducedSpacing,
      pluginState,
      disabled,
    } = this.props;

    return (
      <Wrapper>
        <Dropdown
          mountTo={popupsMountPoint}
          boundariesElement={popupsBoundariesElement}
          scrollableElement={popupsScrollableElement}
          isOpen={this.state.isOpen}
          handleClickOutside={this.hide}
          handleEscapeKeydown={this.hide}
          fitWidth={242}
          fitHeight={80}
          trigger={
            <ToolbarButton
              spacing={isReducedSpacing ? 'none' : 'default'}
              disabled={disabled}
              selected={isOpen}
              title="Text alignment"
              aria-label="Text alignment"
              className="align-btn"
              onClick={this.toggleOpen}
              iconBefore={
                <TriggerWrapper>
                  {iconMap[pluginState.align]}
                  <ExpandIconWrapper>
                    <ExpandIcon label={'Alignment'} />
                  </ExpandIconWrapper>
                </TriggerWrapper>
              }
            />
          }
        >
          <Alignment
            onClick={align => this.changeAlignment(align)}
            selectedAlignment={pluginState.align}
          />
        </Dropdown>
        <Separator />
      </Wrapper>
    );
  }

  private changeAlignment = (align: AlignmentState) => {
    this.toggleOpen();
    return this.props.changeAlignment(align);
  };

  private toggleOpen = () => {
    this.handleOpenChange({ isOpen: !this.state.isOpen });
  };

  private handleOpenChange = ({ isOpen }: { isOpen: boolean }) => {
    this.setState({ isOpen });
  };

  private hide = () => {
    if (this.state.isOpen === true) {
      this.setState({ isOpen: false });
    }
  };
}

export default AlignmentToolbar;
