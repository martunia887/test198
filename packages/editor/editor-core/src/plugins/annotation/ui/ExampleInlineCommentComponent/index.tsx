import * as React from 'react';

import Avatar, { AvatarItem } from '@atlaskit/avatar';
import Button from '@atlaskit/button';
import { Popup } from '@atlaskit/editor-common';
import EditorRemoveIcon from '@atlaskit/icon/glyph/editor/remove';
import TextArea from '@atlaskit/textarea';

import { uuid } from '../../../../utils/input-rules';
import { InlineCommentComponentProps } from '../../types';

const getInlineCommentId = () => `inline-comment-${uuid()}`;

export type ExampleInlineCommentComponentState = {
  commentingValue: string;
};

export default class ExampleInlineCommentComponent extends React.Component<
  InlineCommentComponentProps,
  ExampleInlineCommentComponentState
> {
  state = {
    commentingValue: '',
  };

  componentWillMount() {
    this.setState({
      commentingValue: '',
    });
  }

  shouldComponentUpdate(
    nextProps: InlineCommentComponentProps,
    nextState: ExampleInlineCommentComponentState,
  ) {
    return (
      this.props.dom !== nextProps.dom ||
      this.state.commentingValue !== nextState.commentingValue
    );
  }

  componentDidUpdate(prevProps: InlineCommentComponentProps) {
    const { markerRef, onCancel } = this.props;

    /*
      Since the Editor loses focus when the actual comment is being made,
      we need to add a temporary mark in ProseMirror to show the selection 
      to the user. If the user dismisses the comment creation, the query mark 
      must be removed in the Editor using this callback
    */
    if (!markerRef && prevProps.dom && !this.props.dom) {
      onCancel();
    }
  }

  saveComment = () => {
    const { onSuccess } = this.props;
    const { commentingValue } = this.state;

    /* 
      Component is responsible for creating the inline comment (sync or async),
      then passes the IC's identifier back to the Editor using onSuccess()
    */
    const id = getInlineCommentId();

    window.localStorage.setItem(id, commentingValue);
    this.setState({ commentingValue: '' });

    onSuccess(id);
  };

  renderInsertComment = () => {
    const { commentingValue } = this.state;

    return (
      <div
        style={{
          padding: '10px',
        }}
      >
        <div style={{ paddingBottom: '10px' }}>
          <TextArea
            onChange={(event: React.ChangeEvent<HTMLTextAreaElement>) =>
              this.setState({ commentingValue: event.target.value })
            }
            value={commentingValue}
          />
        </div>
        <Button appearance="primary" onClick={this.saveComment}>
          Save
        </Button>
      </div>
    );
  };

  onDelete = () => {
    const { markerRef, onDelete } = this.props;

    if (markerRef) {
      window.localStorage.removeItem(markerRef);
      onDelete(markerRef);
    }
  };

  getCommentValue(id: string) {
    return (
      window.localStorage.getItem(id) ||
      'Easy! This is just a basic inline comments test!'
    );
  }

  renderShowComment(id: string) {
    return (
      <div
        style={{
          padding: '10px',
        }}
      >
        {this.getCommentValue(id)}
        <div style={{ padding: '5px 0' }}>
          <Button appearance="subtle" onClick={this.onDelete} spacing="none">
            <EditorRemoveIcon label="remove comment" />
          </Button>
        </div>
      </div>
    );
  }

  renderContent = () => {
    const { markerRef } = this.props;

    if (markerRef) {
      return this.renderShowComment(markerRef);
    }

    return this.renderInsertComment();
  };

  getStyles(showing: boolean): React.CSSProperties {
    return showing
      ? {
          backgroundColor: 'white',
          minHeight: '100px',
          width: '250px',
          overflow: 'hidden',
          padding: '5px',
          border: '1px solid #e2e2e2',
          transition: '200ms width ease-in',
        }
      : {
          width: 0,
          visibility: 'hidden',
          transition: '200ms width ease-in',
        };
  }

  render() {
    const { dom, markerRef, textSelection } = this.props;

    return (
      <Popup
        target={
          dom || (document.querySelectorAll('.ProseMirror')[0] as HTMLElement)
        }
        alignY="bottom"
        fitHeight={200}
        fitWidth={200}
        alignX={'right'}
        offset={[
          dom
            ? -(window.innerWidth - dom.getBoundingClientRect().right - 50)
            : 0,
          20,
        ]}
      >
        <div style={this.getStyles(!!markerRef || !!textSelection)}>
          <AvatarItem
            avatar={
              <Avatar
                src="https://api.adorable.io/avatars/80/chaki@me.com.png"
                presence="online"
              />
            }
            key={'vsutrave@atlassian.com'}
            primaryText={'Vijay Sutrave'}
            secondaryText={'vsutrave@atlassian.com'}
          />
          {this.renderContent()}
        </div>
      </Popup>
    );
  }
}
