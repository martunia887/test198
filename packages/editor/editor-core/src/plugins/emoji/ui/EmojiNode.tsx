import React, { FunctionComponent } from 'react';
import { Node as ProsemirrorNode } from 'prosemirror-model';
import Emoji from './Emoji';
import { useProviderFactory } from '@atlaskit/editor-common/provider-factory';
import { EmojiAttributes } from '@atlaskit/adf-schema';

interface EmojiNodeProps {
  node: ProsemirrorNode;
}

export const EmojiNode: FunctionComponent<EmojiNodeProps> = ({ node }) => {
  const providerFactory = useProviderFactory();

  const { id, shortName, text } = node.attrs as EmojiAttributes;

  return (
    <Emoji
      providers={providerFactory}
      id={id}
      shortName={shortName}
      fallback={text}
    />
  );
};
