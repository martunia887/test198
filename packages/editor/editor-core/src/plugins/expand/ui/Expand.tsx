import * as React from 'react';
import { defineMessages, injectIntl, InjectedIntlProps } from 'react-intl';
import { EditorView } from 'prosemirror-view';
import { Node as PmNode } from 'prosemirror-model';
import { findParentNodeOfTypeClosestToPos } from 'prosemirror-utils';
import ChevronRightIcon from '@atlaskit/icon/glyph/chevron-right';
import Tooltip from '@atlaskit/tooltip';
import {
  Container,
  Input,
  Icon,
  Content,
  TitleButton,
  TitleContainer,
} from './styles';
import {
  selectExpand,
  updateExpandTitle,
  setCursorInsideExpand,
} from '../commands';

export const messages = defineMessages({
  expandNode: {
    id: 'fabric.editor.expand',
    defaultMessage: 'Expand',
    description: 'Expand the node',
  },
  collapseNode: {
    id: 'fabric.editor.collapse',
    defaultMessage: 'Collapse',
    description: 'Collapse the node',
  },
  expandPlaceholderText: {
    id: 'fabric.editor.expandPlaceholder',
    defaultMessage: 'Give this expand a title...',
    description: 'A placeholder text for expand node title input field',
  },
});

interface Props {
  node: PmNode;
  view: EditorView;
  contentDOMRef: (node: HTMLElement | null) => void;
  pos: number;
}

interface State {
  collapsed: boolean;
}

class Expand extends React.PureComponent<Props & InjectedIntlProps, State> {
  state = {
    collapsed: false,
  };

  render() {
    const {
      node,
      intl: { formatMessage },
    } = this.props;
    const { title } = node.attrs;
    const { collapsed } = this.state;
    const label = this.props.node.type.name;
    const tooltipMessage = formatMessage(
      collapsed ? messages.expandNode : messages.collapseNode,
    );

    return (
      <Container onClick={this.onSelect} collapsed={collapsed}>
        <TitleContainer onClick={this.onContentClick}>
          <TitleButton onClick={this.onCollapse} aria-label={tooltipMessage}>
            <Tooltip content={tooltipMessage} position="top">
              <Icon collapsed={collapsed} data-expand-buton>
                <ChevronRightIcon label={label} />
              </Icon>
            </Tooltip>
          </TitleButton>
          <Input
            type="text"
            value={title}
            placeholder={formatMessage(messages.expandPlaceholderText)}
            onChange={this.onTitleChange}
            onFocus={this.onTitleFocus}
          />
        </TitleContainer>
        {collapsed ? null : (
          <Content
            onClick={this.onContentClick}
            innerRef={ref => {
              this.props.contentDOMRef(ref);
            }}
          />
        )}
      </Container>
    );
  }

  private onContentClick = (event: React.SyntheticEvent) => {
    event.stopPropagation();
  };

  private onTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { state, dispatch } = this.props.view;
    const pos = this.getPosFromInput(event.target);
    updateExpandTitle(event.target.value, pos, this.props.node.type)(
      state,
      dispatch,
    );
  };

  private onTitleFocus = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { view } = this.props;
    const { tr } = view.state;
    const pos = this.getPosFromInput(event.target);
    view.dispatch(setCursorInsideExpand(pos, tr, 1));
  };

  private onCollapse = () => {
    this.setState({ collapsed: !this.state.collapsed });
  };

  private onSelect = () => {
    const { state, dispatch } = this.props.view;
    selectExpand(this.props.pos)(state, dispatch);
  };

  private getPosFromInput = (target: HTMLElement): number => {
    const { schema, doc } = this.props.view.state;
    const pos = this.props.view.posAtDOM(target, 0);
    const node = findParentNodeOfTypeClosestToPos(doc.resolve(pos), [
      schema.nodes.nestedExpand,
      schema.nodes.expand,
    ]);
    return node ? node.pos : this.props.pos;
  };
}

export default injectIntl(Expand);
