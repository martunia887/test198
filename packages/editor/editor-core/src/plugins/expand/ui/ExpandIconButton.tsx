import React, { useCallback } from 'react';
import Button from '@atlaskit/button';
import {
  akEditorSwoopCubicBezier,
  expandMessages,
  ExpandTooltipWrapper,
} from '@atlaskit/editor-common';
import ChevronRightIcon from '@atlaskit/icon/glyph/chevron-right';
import { colors } from '@atlaskit/theme';
import Tooltip from '@atlaskit/tooltip';
import { InjectedIntl } from 'react-intl';
import { expandClassNames } from './class-names';

interface ExpandIconButtonProps {
  expanded: boolean;
  intl?: InjectedIntl;
}

export const ExpandIconButton = (props: ExpandIconButtonProps) => {
  const { expanded, intl } = props;
  const message = expanded
    ? expandMessages.collapseNode
    : expandMessages.expandNode;
  const label = (intl && intl.formatMessage(message)) || message.defaultMessage;

  const useTheme = useCallback(
    (currentTheme: any, themeProps: any) => {
      const { buttonStyles, ...rest } = currentTheme(themeProps);
      return {
        buttonStyles: {
          ...buttonStyles,
          height: '100%',
          '& svg': {
            transform: props.expanded
              ? 'transform: rotate(90deg);'
              : 'tranform: rotate(0deg);',
            transition: `transform 0.2s ${akEditorSwoopCubicBezier};`,
          },
        },
        ...rest,
      };
    },
    [props],
  );

  return (
    <Tooltip content={label} position="top" tag={ExpandTooltipWrapper}>
      <Button
        appearance="subtle"
        className={expandClassNames.iconContainer}
        iconBefore={
          <ChevronRightIcon label={label} primaryColor={colors.N80A} />
        }
        shouldFitContainer
        theme={useTheme}
      ></Button>
    </Tooltip>
  );
};
