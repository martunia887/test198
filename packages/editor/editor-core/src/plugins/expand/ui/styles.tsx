import styled, { css } from 'styled-components';
import { colors, gridSize, fontSize } from '@atlaskit/theme';
import {
  blockNodesVerticalMargin,
  akEditorExpandLayoutOffset,
} from '@atlaskit/editor-common';

export const Input = styled.input`
  outline: none;
  border: none;
  font-size: ${fontSize}px;
  line-height: 1.714;
  font-weight: normal;
  color: ${colors.N200};
  background: transparent;
  display: flex;
  flex: 1;
  padding: 0 0 0 ${gridSize() / 2}px;

  &::placeholder {
    opacity: 0.6;
  }
`;

export const Content = styled.div`
  cursor: text;
`;

export const expandStyles = css`
  .expandView-content-wrap,
  .nestedExpandView-content-wrap {
    margin-top: ${blockNodesVerticalMargin}rem;
  }

  .ProseMirror > .expandView-content-wrap {
    margin-left: -${akEditorExpandLayoutOffset}px;
    margin-right: -${akEditorExpandLayoutOffset}px;
  }
`;
