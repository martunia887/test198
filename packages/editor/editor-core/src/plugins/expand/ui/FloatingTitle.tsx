import * as React from 'react';
import { injectIntl, InjectedIntlProps } from 'react-intl';
import { Node as PMNode } from 'prosemirror-model';
import { EditorView } from 'prosemirror-view';
import { isNodeSelection } from 'prosemirror-utils';
import { Popup } from '@atlaskit/editor-common';
import { gridSize } from '@atlaskit/theme';
import { Input } from './styles';
import { messages } from './Expand';
import { updateExpandTitle, setShouldFocusTitle } from '../commands';

interface Props {
  view: EditorView;
  expandRef?: HTMLDivElement | null;
  expandNode?: PMNode | null;
  expandPosition?: number;
  parentLayout?: string;
  shouldFocusTitle?: boolean;
  containerWidth?: number;
  mountPoint?: HTMLElement;
  boundariesElement?: HTMLElement;
  scrollableElement?: HTMLElement;
}

interface State {
  title: string;
}

const inputOffset = gridSize() * 6 - gridSize() / 2 + 2;

class FloatingTitle extends React.PureComponent<
  Props & InjectedIntlProps,
  State
> {
  constructor(props: Props & InjectedIntlProps) {
    super(props);

    this.state = {
      title: props.expandNode ? props.expandNode.attrs.title : '',
    };
  }

  componentDidUpdate(prevProps: Props & InjectedIntlProps) {
    if (prevProps.expandNode !== this.props.expandNode) {
      this.setState({
        title: this.props.expandNode ? this.props.expandNode.attrs.title : '',
      });
    }
  }

  render() {
    const {
      intl: { formatMessage },
      expandRef,
      mountPoint,
      scrollableElement,
      expandNode,
      view,
    } = this.props;
    if (!expandNode || !expandRef || isNodeSelection(view.state.selection)) {
      return null;
    }
    const target = expandRef.querySelector(
      '[data-expand-title]',
    ) as HTMLElement;
    if (!target) {
      return null;
    }

    return (
      <Popup
        alignX="left"
        alignY="start"
        target={target}
        mountTo={mountPoint}
        boundariesElement={target}
        scrollableElement={scrollableElement}
        offset={[0, 0.5]}
        forcePlacement
        allowOutOfBounds
      >
        <Input
          type="text"
          innerRef={this.handleRef}
          value={this.state.title}
          width={expandRef.offsetWidth - inputOffset}
          onChange={this.onChange}
          placeholder={formatMessage(messages.expandPlaceholderText)}
        />
      </Popup>
    );
  }

  private handleRef = (ref: HTMLElement | null) => {
    if (ref && this.props.shouldFocusTitle) {
      ref.focus();
      // prevend handling click by GapCursor :0
      setTimeout(() => {
        const { state, dispatch } = this.props.view;
        setShouldFocusTitle(false)(state, dispatch);
      }, 300);
    }
  };

  private onChange = (event: React.SyntheticEvent) => {
    if (!this.props.expandNode) {
      return;
    }
    const { value } = event.target as HTMLInputElement;
    if (this.props.expandNode.attrs.title !== value) {
      this.setState({ title: value }, this.updateTitle);
    }
  };

  private updateTitle = () => {
    const { expandPosition, expandNode } = this.props;
    if (!expandNode || typeof expandPosition !== 'number') {
      return;
    }
    const { state, dispatch } = this.props.view;
    if (expandNode.attrs.title !== this.state.title) {
      updateExpandTitle(this.state.title, expandNode, expandPosition)(
        state,
        dispatch,
      );
    }
  };
}

export default injectIntl(FloatingTitle);
