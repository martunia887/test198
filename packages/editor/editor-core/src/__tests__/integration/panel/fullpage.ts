import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
  insertBlockMenuItem,
  insertMentionUsingClick,
  navigateOrClear,
} from '../_helpers';

import { messages as insertBlockMessages } from '../../../plugins/insert-block/ui/ToolbarInsertBlock';

BrowserTestCase(
  'change-type.ts: Change the type of panel to Error',
  { skip: ['edge', 'ie'] },
  async client => {
    const browser = new Page(client);

    await navigateOrClear(browser, fullpage.path);

    await browser.waitForSelector(fullpage.placeholder);
    await browser.click(fullpage.placeholder);
    await quickInsert(browser, 'Panel');

    await browser.type(editable, 'this text should be in the panel');

    // Change panel type to Error
    const selector = `[aria-label="Error"]`;
    await browser.click(selector);

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'insert-link.ts: Insert link in panel by typing Markdown',
  { skip: ['edge', 'ie'] },
  async client => {
    const browser = new Page(client);

    await navigateOrClear(browser, fullpage.path);

    await browser.waitForSelector(fullpage.placeholder);
    await browser.click(fullpage.placeholder);

    await quickInsert(browser, 'Panel');

    await browser.type(editable, '[Atlassian](https://www.atlassian.com/)');

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'insert-toolbar-menu.ts: Insert panel via toolbar menu',
  { skip: ['ie'] },
  async client => {
    const browser = new Page(client);

    await navigateOrClear(browser, fullpage.path);
    await browser.waitForSelector(editable);
    await browser.click(editable);

    await insertBlockMenuItem(browser, 'Panel');

    await browser.type(editable, 'this text should be in the panel');

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'inside-table.ts: Insert panel into table, add text, change panel type',
  { skip: ['edge', 'ie'] },
  async client => {
    const insertTableMenu = `[aria-label="${
      insertBlockMessages.table.defaultMessage
    }"]`;
    const tableControls = '[aria-label="Table floating controls"]';

    const browser = new Page(client);

    await navigateOrClear(browser, fullpage.path);
    await browser.waitForSelector(editable);
    await browser.click(editable);
    await browser.click(insertTableMenu);
    await browser.waitForSelector(tableControls);

    await quickInsert(browser, 'Panel');

    // type some text
    await browser.type(editable, 'this text should be in the panel');

    // click on Error label
    const selector = `[aria-label="Error"]`;
    await browser.click(selector);

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
    expect(await browser.isExisting(tableControls)).toBe(false);
    expect(await browser.isVisible(tableControls)).toBe(false);
  },
);

BrowserTestCase(
  'mention.ts: Can insert mention inside panel using click',
  { skip: ['ie', 'edge', 'safari'] },
  async client => {
    const browser = new Page(client);

    await navigateOrClear(browser, fullpage.path);
    await browser.waitFor(editable);
    await browser.click(editable);
    await quickInsert(browser, 'Panel');

    await insertMentionUsingClick(browser, '0');
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'quick-insert.ts: Insert panel via quick insert',
  { skip: ['edge', 'ie'] },
  async client => {
    const browser = new Page(client);

    await navigateOrClear(browser, fullpage.path);
    await browser.waitForSelector(editable);
    await browser.click(editable);
    await quickInsert(browser, 'Panel');

    await browser.type(editable, 'this text should be in the panel');

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);
