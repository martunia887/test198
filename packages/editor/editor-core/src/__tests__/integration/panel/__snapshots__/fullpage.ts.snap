// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change-type.ts: Change the type of panel to Error 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "error",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this text should be in the panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`insert-link.ts: Insert link in panel by typing Markdown 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "marks": Array [
                Object {
                  "attrs": Object {
                    "__confluenceMetadata": null,
                    "href": "https://www.atlassian.com/",
                  },
                  "type": "link",
                },
              ],
              "text": "Atlassian",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`insert-toolbar-menu.ts: Insert panel via toolbar menu 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this text should be in the panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;

exports[`inside-table.ts: Insert panel into table, add text, change panel type 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "panelType": "error",
                  },
                  "content": Array [
                    Object {
                      "content": Array [
                        Object {
                          "text": "this text should be in the panel",
                          "type": "text",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "type": "panel",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;


exports[`mention.ts: Can insert mention inside panel using click 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "accessLevel": "",
                "id": "0",
                "text": "@Carolyn",
                "userType": null,
              },
              "type": "mention",
            },
            Object {
              "text": " ",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;


exports[`quick-insert.ts: Insert panel via quick insert 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "info",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this text should be in the panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
}
`;
